File: Dialogs/DialogViewModelDemo/app/src/main/java/edu/cs4730/dialogviewmodeldemo/myEditNameDialogFrag.java
package edu.cs4730.dialogviewmodeldemo;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.view.ContextThemeWrapper;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.ViewModelProvider;

import edu.cs4730.dialogviewmodeldemo.databinding.FragmentEditNameBinding;

/**
 * This is a "custom" dialog, which has an edittext box and returns
 * the value back via a listener.
 * this also pops up the keyboard for the editext box as well.
 */

public class myEditNameDialogFrag extends DialogFragment {

    public myEditNameDialogFrag() {
        // Empty constructor required for DialogFragment
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        LayoutInflater inflater = getLayoutInflater();
        FragmentEditNameBinding binding = FragmentEditNameBinding.inflate(inflater);

        //the keyboard should come up and focus should be set to this input box.
        binding.txtYourName.requestFocus();

        AlertDialog.Builder builder = new AlertDialog.Builder(new ContextThemeWrapper(requireActivity(), R.style.ThemeOverlay_AppCompat_Dialog));
        builder.setView(binding.getRoot()).setTitle("Hello");
        DataViewModel mViewModel = new ViewModelProvider(requireActivity()).get(DataViewModel.class);

        builder.setPositiveButton("Done", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int id) {
                mViewModel.setItem1(binding.txtYourName.getText().toString());
                dismiss();
            }
        }).setCancelable(false);  //don't let them cancel this dialog.  ie use the backbutton to get out of it.

        return builder.create();
    }
}

File: Dialogs/DialogViewModelDemo_kt/app/src/main/java/edu/cs4730/dialogviewmodeldemo_kt/myEditNameDialogFrag.kt
package edu.cs4730.dialogviewmodeldemo_kt

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.view.ContextThemeWrapper
import androidx.fragment.app.DialogFragment
import androidx.lifecycle.ViewModelProvider
import edu.cs4730.dialogviewmodeldemo_kt.databinding.FragmentEditNameDialogBinding

/**
 * This is a "custom" dialog, which has an edittext box and returns
 * the value back via a listener.
 * this also pops up the keyboard for the editext box as well.
 */
class myEditNameDialogFrag : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = layoutInflater
        val binding: FragmentEditNameDialogBinding = FragmentEditNameDialogBinding.inflate(inflater)


        val builder = AlertDialog.Builder(
            ContextThemeWrapper(
                requireActivity(),
                R.style.AppTheme_Dialog  //com.google.android.material.R.style.ThemeOverlay_MaterialComponents_Dialog
            )
        )
        builder.setView(binding.root).setTitle("Hello")
        val mViewModel = ViewModelProvider(requireActivity())[DataViewModel::class.java]
        builder.setPositiveButton(
            "Done"
        ) { dialog, id ->
            mViewModel.setItem1(binding.txtYourName.text.toString())
            dismiss()
        }
            .setCancelable(false) //don't let them cancel this dialog.  ie use the backbutton to get out of it.

        return builder.create()
    }
}
